src/dune_rules/merlin/merlin.ml

```diff
@@ -466,12 +466,20 @@ module Unprocessed = struct
        } as t) sctx ~dir ~more_src_dirs ~expander =
     let open Action_builder.O in
     let+ config =
-      let* stdlib_dir =
+      let* stdlib_dir, extra_obj_dirs =
         Action_builder.of_memo
         @@
         match t.config.mode with
-        | `Ocaml -> Memo.return (Some stdlib_dir)
-        | `Melange -> Melange_binary.where sctx ~loc:None ~dir
+        | `Ocaml -> Memo.return (Some stdlib_dir, [])
+        | `Melange -> (
+          let open Memo.O in
+          let+ dirs = Melange_binary.where sctx ~loc:None ~dir in
+          match dirs with
+          | [] -> (None, [])
+          | [ stdlib_dir ] -> (Some (Path.external_ stdlib_dir), [])
+          | stdlib_dir :: extra_obj_dirs ->
```

Comment:
Is it a little weird that the first directory is being special cased?

Reply:
I agree, and it might not necessarily be the stdlib if we're not careful, I think that might not matter though?